src/System.Buffers.Primitives/System/SpanExtensions.cs

```diff
@@ -38,124 +38,122 @@ public static bool StartsWith<T>(this ReadOnlySpan<T> items, ReadOnlySpan<T> sli
             fixed (byte* pSearchSpace = &buffer.DangerousGetPinnableReference())
             {
-                return IndexOfVectorized(pSearchSpace, buffer.Length, value0, value1);
+                return IndexOf(pSearchSpace, buffer.Length, value0, value1);
             }
         }
 
-        public unsafe static int IndexOfVectorized(this Span<byte> buffer, byte value0, byte value1, byte value2)
+        public unsafe static int IndexOf(this Span<byte> buffer, byte value0, byte value1, byte value2)
         {
             fixed (byte* pSearchSpace = &buffer.DangerousGetPinnableReference())
             {
-                return IndexOfVectorized(pSearchSpace, buffer.Length, value0, value1, value2);
+                return IndexOf(pSearchSpace, buffer.Length, value0, value1, value2);
             }
         }
 
-        public unsafe static int IndexOfVectorized(this ReadOnlySpan<byte> buffer, byte value)
+        public static int SequentialIndexOf(this ReadOnlySpan<byte> span, byte value)
         {
-            fixed (byte* pSearchSpace = &buffer.DangerousGetPinnableReference())
-            {
-                return IndexOfVectorized(pSearchSpace, buffer.Length, value);
-            }
+            return SequentialIndexOf(ref span.DangerousGetPinnableReference(), value, span.Length);
         }
 
-        public unsafe static int IndexOfVectorized(this ReadOnlySpan<byte> buffer, byte value0, byte value1)
+        public unsafe static int IndexOf(this ReadOnlySpan<byte> buffer, byte value0, byte value1)
         {
             fixed (byte* pSearchSpace = &buffer.DangerousGetPinnableReference())
             {
-                return IndexOfVectorized(pSearchSpace, buffer.Length, value0, value1);
+                return IndexOf(pSearchSpace, buffer.Length, value0, value1);
             }
         }
 
-        public unsafe static int IndexOfVectorized(this ReadOnlySpan<byte> buffer, byte value0, byte value1, byte value2)
+        public unsafe static int IndexOf(this ReadOnlySpan<byte> buffer, byte value0, byte value1, byte value2)
         {
             fixed (byte* pSearchSpace = &buffer.DangerousGetPinnableReference())
             {
-                return IndexOfVectorized(pSearchSpace, buffer.Length, value0, value1, value2);
+                return IndexOf(pSearchSpace, buffer.Length, value0, value1, value2);
             }
         }
 
-        private static unsafe int IndexOfVectorized(byte* searchSpace, int length, byte value)
+        private static unsafe int SequentialIndexOf(ref byte searchSpace, byte value, int length)
         {
-            var offset = 0;
-            // If length < vector length the jump over Vector dominates the search; as the Vector section is quite chunky
-            // So do an early search and exit
-            if (length < Vector<byte>.Count)
+            Debug.Assert(length >= 0);
+
+            IntPtr index = (IntPtr)0; // Use IntPtr for arithmetic to avoid unnecessary 64->32->64 truncations
```

Comment:
Similar to above, should this be fixed everywhere?

https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/SpanHelpers.T.cs#L53
https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/SpanHelpers.byte.cs#L55